### PR TITLE
[apidiff] Add q quick'n'dirty script to produce a diff between any two bundle.zip

### PR DESCRIPTION
**Pull request by spouliot**
Use for producing API diff for release notes without waiting for a PR,
bots and/or approvals...

Also useful to produce API diff between any versions, not just between
the current revision and a baseline (last stable).

### CODE REVIEW

File: tools/apidiff/diff-bundles.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+# anything alreaady built is wrong
+make clean
+
+# unzip the base/origin bundle zip
+rm -Rf origin
+unzip -d origin $1
+
+# copy files over (dirty!) the currennt install directories
+cp -r origin/2.1/ ../../_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/lib/mono/2.1/
```
Comment (rolfbjarne) [inline tools/apidiff/diff-bundles.sh]: `cp -cr` will clone files when possible, which is *much* faster.

File: tools/apidiff/diff-bundles.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# ./diff-bundles.sh base.zip target.zip
+
+# anything alreaady built is wrong
+make clean
+
+# unzip the base/origin bundle zip
+rm -Rf origin
+unzip -d origin $1
+
+# copy files over (dirty!) the currennt install directories
```
Comment (rolfbjarne) [inline tools/apidiff/diff-bundles.sh]:
```suggestion
# copy files over (dirty!) the current install directories
```

File: tools/apidiff/diff-bundles.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+cp -r origin/Xamarin.WatchOS/ ../../_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/lib/mono/Xamarin.WatchOS/
+
+cp origin/Xam*.dll* ../../_mac-build/Library/Frameworks/Xamarin.Mac.framework/Versions/git/lib/mono/ 
+cp -r origin/4.5/ ../../_mac-build/Library/Frameworks/Xamarin.Mac.framework/Versions/git/lib/mono/4.5/
+cp -r origin/Xamarin.Mac/ ../../_mac-build/Library/Frameworks/Xamarin.Mac.framework/Versions/git/lib/mono/Xamarin.Mac/
+
+# cleanup after ourselves
+rm -Rf origin
+
+# run Makefile using the target/currennt bundle
```
Comment (rolfbjarne) [inline tools/apidiff/diff-bundles.sh]:
```suggestion
# run Makefile using the target/current bundle
```

Comment (spouliot) [reply tools/apidiff/diff-bundles.sh]: yeah, my laptop 'n' key is broken :(
need to set an appointment to the apple store

**Comment by spouliot** [comment]
* [FAIL] LayerTest.TestBug26532 : disposed text layers -> known random issue -> https://github.com/xamarin/maccore/issues/1595
* [FAIL] NSMutableDictionary2Test.AddEntries : ICE 1  -> random (unrelated) filed as https://github.com/xamarin/maccore/issues/1605

**Pull request merged**